careers/careers/static/js/filters.js

```diff
@@ -114,9 +114,30 @@
          */
         filterPositions: function(field, value) {
             this.$positionTable.find('tr.position').each(function(index, element) {
+                var data = element.dataset[field];
                 if (!value)
                     return;
-                if (element.dataset[field].indexOf(value + ',') === -1) {
+
+                if (data.indexOf(value + ',') === -1) {
+                    element.classList.add('hidden');
+                }
+            });
+        },
+        filterLocations: function(value) {
+            this.$positionTable.find('tr.position').each(function(index, element) {
+                var data = element.dataset.location;
+                if (!value)
+                    return;
+
+                // When user selects 'Remote' only list jobs explicitly marked
+                // Remote otherwise list jobs matching value (which is a mozilla
+                // office) and those marked as 'All Offices'
+                if (value === 'Remote') {
+                    if (data.indexOf(value + ',') === -1) {
```

Comment:
If combined and `value === Remote` but not in `data` then `else if` will be checked which is not the desired functionality.